src/ompl/tools/thunder/src/Thunder.cpp

```diff
@@ -82,8 +84,20 @@ void ompl::tools::Thunder::setup()
 
         // Setup planning from scratch planner
         OMPL_INFORM("Initializing planners");
-        // planner_vec_.clear();
-        // planner_vec_ = {nullptr, nullptr, nullptr, nullptr};
+        // set number of threads if not already set
+        if(n_threads_ == 0) {
+          n_threads_ = std::max(std::thread::hardware_concurrency(), 2u) - 1;
+        }
+        // set the size of the planner vector
+        planner_vec_.clear();
+        if (plan_with_cforest_) {
+          std::vector<base::PlannerPtr> planner_vec {1};
+          planner_vec_ = planner_vec;
+        } else {
+          std::vector<base::PlannerPtr> planner_vec {n_threads_};
+          planner_vec_ = planner_vec;
+        }
```

Comment:
how does this code work at all?

the original `planner_vec_` is a vec of pointers, but you are making a vector of a single int element, where the int is the number of threads.

Reply:
https://stackoverflow.com/questions/11457571/how-to-set-initial-size-of-stdvector/11457629

it creates a vector filled with that number of (uninitialized) pointers

Reply:
you have to use `()` not `{}`. i'm sure this is wrong.

Reply:
you are correct! need `(n_threads)` to correctly make more than 1 thread. I'm not sure why it didn't throw -- but I guess since it's a vector of pointer? I was assigning the address to a size_t??

Reply:
> I'm not sure why it didn't throw

i bet the reason is that i wasn't working

Reply:
@jackievideira  do you think this is why you were seeing a high number of solutions returning when hybridize was false?
Other things might be playing a role too (hybridize = false is not perfect and we return approximate solutions), but this could also be playing a role.

Reply:
more discussion on slack 
https://dexai.slack.com/archives/CPZPT2S2W/p1642537366146500?thread_ts=1642089257.062900&cid=CPZPT2S2W